### ISSUE TITLE
dateaxis ticks are not align(vertical) with date values of series

### ISSUE THREAD

**Issue by BrkCoder** [amcharts/amcharts4#1415]
Hey amcharts 4 team,
I have created a basic chart which its dateaxis start at Sun Jun 30 2019 11:00:00 and end at Mon Jul 01 2019 09:00:00. 
The data points jump always in 30 minutes for each point.
There is strange behavior of this basic chart:
When I hover a data point with date value: 21:00, the dateaxis grid line of 21:00 is offset to this point on the left side.
This design create a confusion for the user & the programmer.
In more details: When I am looking on data points which happens at 21:00, the grid line is not vertical to this point as it should because this point start at 21:00, this is a very basic need for every chart.
I have capture this phenomena as picture and codepen:
The picture:
![image](https://user-images.githubusercontent.com/5408870/60396007-9d42e880-9b44-11e9-9842-1efaf3b5ec39.png)

The codepen which reproduce this problem:
https://codepen.io/Barak/pen/wLydZr?editors=0010 

Please confirm if this issue is:
1. a bug 
2. as designed, if so why and how can I achieved my desired normal behavior.

**Comment by BrkCoder** [comment]
Hey I have occurred a similar issue where the workaround was:
`dateAxis.baseInterval = {
  "timeUnit": "second",
  "count": 1
};`

I don't think this is a good workaround why the chart by default take my timestamps which jump evey 30 minutes and divided this equal along the chart and sync with current dateaxis without any strange behaviors such as above!

**Comment by zeroin** [comment]
Date axis shows data items in the middle of the baseInterval. So when your base interval is detected automatically, (30 min), it shows data at 21:15. To change this behavior you must set location of a dateX of a data item:

`series.dataItems.template.locations.dateX = 0;`

Next thing you might want is to change location of axis tooltip, and this should be done by setting 

`dateAxis.renderer.tooltipLocation = 0;`

But I now see that this results some strange tooltip behavior in this particular case, I am looking into it and will let you know.

**Comment by zeroin** [comment]
OK, so a temporary tooltip issue workaround would be this:

`dateAxis.renderer.tooltipLocation = 0.001;
dateAxis.tooltipPosition = "pointer";`

We will try to solve this issue in next release.

**Comment by BrkCoder** [comment]
> Date axis shows data items in the middle of the baseInterval. So when your base interval is detected automatically, (30 min), it shows data at 21:15. To change this behavior you must set location of a dateX of a data item:
> 
> `series.dataItems.template.locations.dateX = 0;`
> 
> Next thing you might want is to change location of axis tooltip, and this should be done by setting
> 
> `dateAxis.renderer.tooltipLocation = 0;`
> 
> But I now see that this results some strange tooltip behavior in this particular case, I am looking into it and will let you know.

Excellent on more thing can you tell me how can I translate this row into a json based config properties:
series.dataItems.template.locations.dateX = 0;
we just used your json based config formula instead of typescript.

**Comment by zeroin** [comment]
Now when I look at it, it seems that it is not possible to do it with JSON only (our JSON guy is off the grid for several days so I have to wait for him to confirm this). Meanwhile, you can simply do this after you create chart with JSON:

`chart.series.getIndex(0).dataItems.template.locations.dateX = 0;`

**Comment by zeroin** [comment]
@martynasma Here is a codepen with JSON:
https://codepen.io/team/amcharts/pen/BgrymE?editors=0010

**Comment by BrkCoder** [comment]
@zeroin thank you very much,
it appears to me that this line only solve all the problem:
`chart.series.getIndex(0).dataItems.template.locations.dateX = 0;`
When you said bug, do you mean that this behavior is classify as a bug?
or the bug was the strange behavior of the tooltip?

**Comment by zeroin** [comment]
Yes, it's the strange tooltip behavior.

**Comment by martynasma** [comment]
We're looking into why the JSON setup does not work.

Meanwhile, this workaround should work:

```
  "series": [{
    "id": "s1",
    "type": "LineSeries",
    "dataFields": {
      "dateX": "date",
      "valueY": "MEASURED"
    },
    "bullets": [{ "type": "CircleBullet" }],
    "tooltipText": "{valueY.value}",
    "callback": function() {
      this.dataItems.template.locations.dateX = 0;
    }
  }],
```

**Comment by martynasma** [comment]
Fixed in [4.5.4](https://github.com/amcharts/amcharts4/releases/tag/4.5.4).

## [4.5.4] - 2019-07-08

### Added
- `allwaysShowTooltip` property added to `Sprite`. Allows tooltip to be shown without the need for hovering/selecting an element.
- `NumberFormatter` now supports `"!"` modifier which works in conjunction with `"a"` and `"b"` modifiers. If used, formatter will force application of a prefix/suffix for values that do not fit into any of the defined prefix lists. This is useful in cases you want to format all of the values using same prefix, e.g. "B" and want even small values to be formatted as such (`500,000` formatted as `"0.5B"`).

### Fixed
- `CategoryAxis` tooltip could show incorrect category if its `renderer.tooltipLocation` was set to `0`.
- Memory leak which happened when disposing a chart with `XYChartScrollbar` with series added to it fixed.
- Some series stacking issues fixed.
- Issue with `baseDuration.timeUnit = "yyyy"` and `StepLineSeries` fixed.
- In some cases `DateAxis` tooltips would show empty near edge of the plot area.
- `CategoryAxis.indexToPosition` adjusted to work better with no default `startLocation` and `endLocation` values.
- `WordCloud` could go into stackoverflow if `minWordLenght` was set to a number which resulted in no words to appear at all.
- `ForceDirectedSeries` were resulting in JS error if inited in a hidden container.
- If an axis range was made invisible by setting its `visible = false`, and then visibility was set back to `true` again, axis range was not shown.
- JSON: entries of type `OrderedListTemplate` or `SortedListTemplate` was not being processed correctly as templated lists.
- Legend items were still focusable/togglable via keyboard even if `clickable`/`focusable` on their respective containers were set to to `false`.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
